src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -252,6 +256,15 @@ protected Query doToQuery(QueryShardContext context) {
             );
         }
 
+        byte[] byteVector = new byte[0];
```

Comment:
why set this to empty array?

Reply:
We need to initialize it or else it is not letting us use it while building the QueryRequest because the logic to add values into this array is inside an if statement.